### ISSUE TITLE
Publish Filtering `+` and Date Filtering

### ISSUE THREAD

**Issue by jonmifsud** [symphonycms/symphony-2#2581]
As you're well aware the Publish Filtering setup; was in concept devised to mimic the same filtering which we have available on the frontend. This seems to work just fine except for this 1 use case which I found:

I'm trying to use the Date Filtering whereby I'm looking to get all entries published on a particular date; When a single date is available and you're using the `datetime` field it doesn't catch the full-day's range which is OK, but for that we would usually use the following filter: `today to today + 1`.

The plus sign however has a 'special' [attribute/meaning when using publish filtering](https://github.com/symphonycms/symphony-2/blob/master/symphony/content/content.publish.php#L391), which means it 'splits' my query into 2 parts which obviously does not work.

@nitriques any idea why this + sign works here and not within the Datasource Filtering? And any particular reason why it is there or  it's being used. I recall it might have come up in earlier conversations. Will be happy to patch this one myself but want to check any ideas/feedback there might be in the community.

**Comment by nitriques** [comment]
Well, the code that does the same thing as the line you mention [is this one](https://github.com/symphonycms/symphony-2/blob/master/symphony/lib/toolkit/class.datasource.php#L219) which is in fact different.

But, the thing I would look into would be the fact that the + character is converted to space, since we pass the value in the query string (my test yeilds `?filter[date]=today%20to%20today%20+%201`)

If I do `window.encodeURIComponent('+')` I get `"%2B"` and ``?filter[date]=today%20to%20today%20%2B%201` but it does not seems to work (or am I out of sync with 2.7.x ?)

**Comment by jonmifsud** [comment]
@nitriques you're right I've encoded the `+` for testing myself; I was confused regarding the latter that even when encoding the `+` it does not seem to work.

I did manage to get to the line you mention, which is the `determine filter type` however the split seems to happen only on the publish filtering. Maybe it should instead be moved on a per-field level, or maybe it already is? I guess each field can determine weather a + is supposed to mean `and` or something else.

**Comment by jonmifsud** [comment]
@nitriques actually if I remove the space and do `+1 day` the filter works. Was doing some testing after looking at the regex properly. However the issue you mentioned to encode `+` to `%2B` still exists, so not going to close this ticket just yet.

**Comment by nitriques** [comment]
Ok great! At least we have a "fix" !?!
